### PR TITLE
feat:implement calcite style 'levenshtein' string function

### PR DESCRIPTION
**Pull request by Syleechan**
## Which issue does this PR close?

Closes #.

## Rationale for this change

https://calcite.apache.org/docs/reference.html#:~:text=LEVENSHTEIN(string1%2C%20string2,string1%20and%20string2

### CODE REVIEW

File: datafusion/physical-expr/src/string_expressions.rs [hunk 1]
```diff
@@ -643,6 +644,31 @@ pub fn overlay<T: OffsetSizeTrait>(args: &[ArrayRef]) -> Result<ArrayRef> {
     }
 }
 
+///Returns the Levenshtein distance between the two given strings
+/// LEVENSHTEIN('kitten', 'sitting') = 3
+pub fn levenshtein<T: OffsetSizeTrait>(args: &[ArrayRef]) -> Result<ArrayRef> {
```
Comment (alamb) [inline datafusion/physical-expr/src/string_expressions.rs:649]: Would it be possible to use the existing implementation in : https://github.com/apache/arrow-datafusion/blob/aaaf698b5e46dd1c4d29bce69dc539a667d2dda6/datafusion/common/src/utils.rs#L484

File: datafusion/physical-expr/Cargo.toml [hunk 2]
```diff
@@ -67,6 +67,7 @@ regex = { version = "1.8", optional = true }
 sha2 = { version = "^0.10.1", optional = true }
 unicode-segmentation = { version = "^1.7.1", optional = true }
 uuid = { version = "^1.2", features = ["v4"] }
+edit-distance = "2.1.0"
```
Comment (alamb) [inline datafusion/physical-expr/Cargo.toml:70]: I think we can avoid this new dependency by either reusing our existing implementation (see below)

**Review by alamb — COMMENTED** [review]
Thank you for the contribution @Syleechan . I think we can avoid the new dependency but otherwise this looks great

**Comment by Syleechan** [comment]
retest this please

**Comment by Syleechan** [comment]
@alamb thanks, I have changed to use exsiting implementation. But the code checks maybe something wrong, I have used the checking command to run local all have passed.

**Comment by alamb** [comment]
I believe the failures are due to some logical conflicts since fixed in https://github.com/apache/arrow-datafusion/pull/8187

Updating to the latest main should work I think

**Comment by Syleechan** [comment]
@alamb thanks, it works. And wait for your code review.

**Review by alamb — APPROVED** [review]
looks good to me -- thank you @Syleechan

**Review by Ted-Jiang — APPROVED** [review]

**Pull request merged**